### PR TITLE
feat(composer): add overlay panel config editor

### PR DESCRIPTION
**Pull request by sheilaXu**
## Overview
Add checkbox to control `isPinned` value for data overlay panel

overlay-panel-config-editor

### Scene Composer
For `scene-composer` package changes specifically, you can preview the component in the published storybook artifact. To do this, wait for the `Publish Storybook` action to complete below.

- Click on the workflow details
- Select the Summary item on the left
- Download the zip file

To run the storybook build locally, you need a local static web server:

```
npm install -g httpserver
cd <Extracted Zip Directory>
httpserver
```

Then open the website http://localhost:8080 to run the doc site.

## Legal
This project is available under the [Apache 2.0 License](http://www.apache.org/licenses/LICENSE-2.0.html).

### CODE REVIEW

File: packages/scene-composer/src/components/panels/scene-components/data-overlay/DataOverlayPanelConfigEditor.tsx [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  const onPinnedChange = useCallback(
+    ({ detail: { checked } }) => {
+      onUpdateCallback({ config: { isPinned: checked } }, false);
+    },
+    [onUpdateCallback],
+  );
+
+  return (
+    <FormField label={formatMessage({ defaultMessage: 'Panel Config', description: 'FormField label' })}>
+      <Checkbox data-testid='pinned-checkbox' checked={config?.isPinned === true} onChange={onPinnedChange}>
```
Comment (divya-sea) [inline packages/scene-composer/src/components/panels/scene-components/data-overlay/DataOverlayPanelConfigEditor.tsx:28]: You don't need to explicitly test for true.

Comment (divya-sea) [inline packages/scene-composer/src/components/panels/scene-components/data-overlay/DataOverlayPanelConfigEditor.tsx:28]: I think you don't need to check explicitly here. You can just say `config.isSpinned`

File: packages/scene-composer/src/components/panels/scene-components/data-overlay/DataOverlayPanelConfigEditor.tsx [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+  const onPinnedChange = useCallback(
+    ({ detail: { checked } }) => {
+      onUpdateCallback({ config: { isPinned: checked } }, false);
+    },
+    [onUpdateCallback],
+  );
+
+  return (
+    <FormField label={formatMessage({ defaultMessage: 'Panel Config', description: 'FormField label' })}>
```
Comment (divya-sea) [inline packages/scene-composer/src/components/panels/scene-components/data-overlay/DataOverlayPanelConfigEditor.tsx:27]: FormField labels are inconsistent throughout the package. Are these verified with Tech writer later?

File: packages/scene-composer/src/components/panels/scene-components/data-overlay/DataOverlayPanelConfigEditor.tsx (hunk 1 above)
Comment (sheilaXu) [reply packages/scene-composer/src/components/panels/scene-components/data-overlay/DataOverlayPanelConfigEditor.tsx:28]: `config.isSpinned` may be undefined so I will get type error

File: packages/scene-composer/src/components/panels/scene-components/data-overlay/DataOverlayPanelConfigEditor.tsx (hunk 2 above)
Comment (sheilaXu) [reply packages/scene-composer/src/components/panels/scene-components/data-overlay/DataOverlayPanelConfigEditor.tsx:27]: I will review with UX later, I don't have any mock yet right now, so I'm basically just making up with an UI that let me input all supported configurations

**Review by divya-sea — APPROVED** [review]

**Pull request merged**